kibana/client.go

```diff
@@ -191,7 +191,14 @@ func NewClientWithConfigDefault(config *ClientConfig, defaultPort int, binaryNam
 		binaryName = "Libbeat"
 	}
 	userAgent := useragent.UserAgent(binaryName, version, commit, buildtime)
-	rt, err := config.Transport.Client(httpcommon.WithHeaderRoundTripper(map[string]string{"User-Agent": userAgent}))
+
+	const elasticAPIVersionHeaderKey = "Elastic-Api-Version"
+	const elasticAPIDefaultVersion = "2023-10-31"
```

Comment:
Could we just add "Elastic-Api-Version" to the `Headers` in `DefaultClientConfig()` function?  That way users can override if they need to for some reason and the defaults would be in the `DefaultClientConfig()` function where the other defaults of localhost & http are.

Reply:
We could... even if now this is the only version that is supported :sweat_smile: 
Will change it